Source/EasyNetQ/RabbitAdvancedBus.cs

```diff
@@ -608,49 +597,52 @@ public async Task ExchangeDeclarePassiveAsync(string name, CancellationToken can
                 );
             }
 
-            return new Binding(destination, source, routingKey, arguments);
+            return new Binding<Exchange>(source, destination, routingKey, arguments);
         }
 
         /// <inheritdoc />
-        public virtual async Task UnbindAsync(IBinding binding, CancellationToken cancellationToken)
+        public virtual async Task UnbindAsync(Binding<Queue> binding, CancellationToken cancellationToken)
         {
-            Preconditions.CheckNotNull(binding, "binding");
-
             using var cts = cancellationToken.WithTimeout(configuration.Timeout);
 
-            if (binding.Bindable is IQueue queue)
-            {
-                await clientCommandDispatcher.InvokeAsync(
-                    x => x.QueueUnbind(queue.Name, binding.Exchange.Name, binding.RoutingKey, null),
-                    cts.Token
-                ).ConfigureAwait(false);
+            await clientCommandDispatcher.InvokeAsync(
+                x => x.QueueUnbind(
+                    binding.Destination.Name, binding.Source.Name, binding.RoutingKey, binding.Arguments?.ToDictionary(e => e.Key, e => e.Value)
```

Comment:
`binding.Arguments?.ToDictionary(e => e.Key, e => e.Value)` 

We have to convert IReadOnlyDictionary to IDictionary

Reply:
same as per `ExchangeUnbind` above